src/Page/Submission/Team1/Algorithm.js

```diff
@@ -0,0 +1,185 @@
+// Sherman Grewal
+// Oliver Lingertat
+// Lucy Cheng
+// Navya Mehta
+import React, { Component } from 'react';
+import { render } from "react-dom";
+import StockStats from '../../../Data/stocks';
+
+class stockobj extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+    keys: Object.keys(StockStats), 
+    dates: StockStats.map(function(x) {
+        return x.map(function(day) {
+          return day.Date;
+        });
+      }),
+    closeVal: StockStats.map(function(x) {
+        return x.map(function(day) {
+          return day.Close;
+        });
+      }),
+    sma19: this.sma19(),
+    ema19: this.ema19(),
+    ema39: this.ema39(),
+    algval: this.mainalg()
+    };
+    this.InitFunction=this.InitFunction.bind(this);
+  }
+
+  InitFunction() {
+    this.props.childHandler(this.state);
+  }
+
+  render() {
+    return (
+        this.InitFunction
+      )
+  }
+
+  sma19() {
+    let closearr = StockStats.map(function(x) {
+        return x.map(function(day) {
+          return day.Close;
```

Comment:
I suggest to use Adjusted Close instead of the close price. In the event of stock split or dividends, the close price might not reflect the company event. However, adjusted close price will take consideration of such event. [FYI](https://www.investopedia.com/terms/a/adjusted_closing_price.asp)